vault/logical_raw.go

```diff
@@ -46,6 +48,17 @@ func NewRawBackend(core *Core) *RawBackend {
 func (b *RawBackend) handleRawRead(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
 	path := data.Get("path").(string)
 
+	// Preserve pre-existing behavior to decompress if `compressed` is missing
+	compressed := true
+	if d, ok := data.GetOk("compressed"); ok {
+		compressed = d.(bool)
+	}
+
+	encoding := data.Get("encoding").(string)
+	if encoding != "" && encoding != "base64" {
+		return logical.ErrorResponse("invalid encoding '%s'", encoding), logical.ErrInvalidRequest
```

Comment:
This is fine, but FYI more commonly we omit the second return (`error`) value when we're using logical.ErrorResponse.  logical.ErrInvalidRequest translates to http.StatusBadRequest, which is the default for when we return an error with a non-nil response, as is the case for logical.ErrorResponse.